br/pkg/lightning/mydump/region.go

```diff
@@ -442,23 +478,47 @@ func SplitLargeFile(
 				pos = dataFile.FileMeta.FileSize
 			}
 			endOffset = pos
-			parser.Close()
+			subParser.Close()
+		}
+		isChunkNeeded := true
+		var rowsToSkip int64
+		// iterate the overall parser to get how many lines to skip in this chunk
+		if remainingSkipRows > 0 {
+			iterPos, _ := parser.Pos()
+			for remainingSkipRows > 0 && iterPos < endOffset {
+				if err := parser.ReadRow(); err != nil {
+					if !errors.ErrorEqual(err, io.EOF) {
+						return 0, nil, nil, err
+					}
+				}
+				iterPos, _ = parser.Pos()
+				remainingSkipRows--
+				rowsToSkip++
+			}
+			if remainingSkipRows <= 0 && iterPos < endOffset {
+				isChunkNeeded = true
+			} else {
+				isChunkNeeded = false
+			}
+		}
+		if isChunkNeeded {
+			regions = append(regions,
+				&TableRegion{
+					DB:       meta.DB,
+					Table:    meta.Name,
+					FileMeta: dataFile.FileMeta,
+					Chunk: Chunk{
+						Offset:         startOffset,
+						EndOffset:      endOffset,
+						PrevRowIDMax:   prevRowIdxMax,
+						RowIDMax:       rowIDMax,
+						Columns:        columns,
+						SkipFirstNRows: rowsToSkip,
```

Comment:
changing `startOffset` is enough? no need to add `SkipFirstNRows`

Reply:
I've considered this problem.  We need the `SkipFirstNRows`, because for some data format like parquet, it is not stored row by row.  In this situation, we cannot just specify a offset to skip first N rows.